File: db4o.net/Db4objects.Db4o.CS/Db4objects.Db4o.CS/CS/Internal/LazyClientObjectSetStub.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Foundation;
using Db4objects.Db4o.Internal.Query.Result;

namespace Db4objects.Db4o.CS.Internal
{
	/// <exclude></exclude>
	public class LazyClientObjectSetStub
	{
		private readonly AbstractQueryResult _queryResult;

		private IIntIterator4 _idIterator;

		public LazyClientObjectSetStub(AbstractQueryResult queryResult, IIntIterator4 idIterator
			)
		{
			_queryResult = queryResult;
			_idIterator = idIterator;
		}

		public virtual IIntIterator4 IdIterator()
		{
			return _idIterator;
		}

		public virtual AbstractQueryResult QueryResult()
		{
			return _queryResult;
		}

		public virtual void Reset()
		{
			_idIterator = _queryResult.IterateIDs();
		}
	}
}

File: db4o.cs/src/com/db4o/cs/internal/LazyClientObjectSetStub.java
/* Copyright (C) 2006  Versant Inc.  http://www.db4o.com */

package com.db4o.cs.internal;

import com.db4o.foundation.*;
import com.db4o.internal.query.result.*;


/**
 * @exclude
 */
public class LazyClientObjectSetStub {
	
	private final AbstractQueryResult _queryResult;
	
	private IntIterator4 _idIterator;
	
	public LazyClientObjectSetStub(AbstractQueryResult queryResult, IntIterator4 idIterator) {
		_queryResult = queryResult;
		_idIterator = idIterator;
	}
	
	public IntIterator4 idIterator(){
		return _idIterator;
	}
	
	public AbstractQueryResult queryResult(){
		return _queryResult;
	}
	
	public void reset(){
		_idIterator = _queryResult.iterateIDs();
	}

}
